packages/headless/src/api/knowledge/answer-slice.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  BaseQueryFn,
+  createApi,
+  FetchArgs,
+  fetchBaseQuery,
+  FetchBaseQueryError,
+  retry,
+} from '@reduxjs/toolkit/query';
+import {ConfigurationSection} from '../../state/state-sections';
+
+type StateNeededByAnswerSlice = ConfigurationSection;
+
+/**
+ * `dynamicBaseQuery` is passed to the baseQuery of the createApi,
+ * but note that the baseQuery will not be used if a queryFn is provided in the createApi endpoint
+ */
+const dynamicBaseQuery: BaseQueryFn<
+  string | FetchArgs,
+  unknown,
+  FetchBaseQueryError
+> = async (args, api, extraOptions) => {
+  const state = api.getState() as StateNeededByAnswerSlice;
+  const {accessToken, organizationId, platformUrl} = state.configuration;
+  const updatedArgs = {
+    ...(args as FetchArgs),
+    headers: {
+      ...((args as FetchArgs)?.headers || {}),
+      Authorization: `Bearer ${accessToken}`,
+    },
+  };
+  try {
+    const data = fetchBaseQuery({
+      baseUrl: `${platformUrl}/rest/organizations/${organizationId}`,
```

Comment:
A bit unrelated for now, but please do keep in your radar when working on the actual knowledge engine itself: 

We'll want to only support/use organizationEndpoints : https://docs.coveo.com/en/mcc80216/manage-an-organization/organization-endpoints

The code is already there in Headless, it's a matter of configuring it correctly when initializing the engine.